dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java

```diff
@@ -298,8 +298,12 @@ private void resultProcess(ResultSet resultSet) throws Exception {
         String result = JSONUtils.toJsonString(resultJSONArray);
         logger.debug("execute sql : {}", result);
 
-        sendAttachment(sqlParameters.getGroupId(), StringUtils.isNotEmpty(sqlParameters.getTitle()) ? sqlParameters.getTitle() : taskExecutionContext.getTaskName() + " query result sets",
-                JSONUtils.toJsonString(resultJSONArray));
+        try {
+            sendAttachment(sqlParameters.getGroupId(), StringUtils.isNotEmpty(sqlParameters.getTitle()) ? sqlParameters.getTitle() : taskExecutionContext.getTaskName() + " query result sets",
+                    JSONUtils.toJsonString(resultJSONArray));
+        } catch (Exception e) {
```

Comment:
SQLTask queries the class task, fails to send mail, SQLTask will fail. Why did this piece have to be modified?

Reply:
As a user, he just want to execute a sql like `update table set`. But because the mail configuration is not set, the `execute sql error` occurs and the task gets failed while sql has been updated. The reason for the error is just `send mail failed`.
So it's weird about this. It would be a more reasonable way to give a warning about the `send mail failed` exception.

Reply:
A few days ago, Chenxingchun and I tested it and found the `execute sql error`, and we both felt weird besides the issue #4621

Reply:
Question 1: SQLTask page has set the query type, 1. When selecting the query, it is to execute the SELECT statement, and the query result is sent by mail, so the page must fill in the relevant parameters of the mail; 2. When you select a non-query type, you are executing INSERT/UPDATE statements, and the relevant parameters of the message will not be displayed on the page. The current change is to catch mail sending exceptions. What is the problem?
The query type is selected in page #4621. If the query type is selected, the query result will be generated. The query result needs to be sent by email, so whether the email is sent successfully also determines the final task status of SQLTask
There are some problems with this design, but I don't think it should be solved by catching email sending exceptions.

---
我是存在几点疑惑的，问题1:sqlTask页面有设置查询类型，1.选择查询时，是执行select语句，查询结果通过邮件发送，所以页面必须填写邮件相关参数；2.选择非查询类型时，是执行insert/update等语句的，此时页面不会展示邮件的相关参数。现在的改动是捕获邮件发送异常，没有想明白解决的是什么问题？

#4621 页面选择的是查询类型，选择查询类型就会有查询结果，查询结果是需要通过邮件发送出去，所以邮件发送是否成功也是决定sqltask最终的任务状态

此次设计是存在一些问题，但是我认为不应该通过捕获邮件发送异常来解决。